### PR TITLE
Add Google Analytics if settings supply it

### PR DESCRIPTION
**Pull request by sean-roberts**
Part of the work for: hypothesis/product-backlog#117

### CODE REVIEW

File: src/sidebar/app.js [hunk 1]
```diff
@@ -37,6 +37,26 @@ if (settings.raven) {
   angular.module('ngRaven', []);
 }
 
+if(settings.googleAnalytics){
```
Comment (robertknight) [inline src/sidebar/app.js]: I've tested this and verified that analytics reporting works in the context of the browser extension. Can we move this to a function in a separate module (eg. `analytics.js`) and just add a note that the minified block at the top is the minified GA tracking snippet.

File: src/sidebar/app.js [hunk 2]
```diff
@@ -37,6 +38,10 @@ if (settings.raven) {
...
```
Comment (robertknight) [inline src/sidebar/app.js]: Nit: There should be spaces around the condition of an if statement, ie. `if (cond) {`. I thought this was included in the ESLint config, apparently that needs adding.

File: src/sidebar/app.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 var queryString = require('query-string');
+var addAnalytics = require('./ga');
```
Comment (robertknight) [inline src/sidebar/app.js]: Nit. Local modules should be grouped separately from npm dependencies:

```
var thing = require('npm-dep');
var thing2 = require('npm-dep2');

var localThing = require('./local-module');
```

Our Python code follows the same convention.

**Comment by robertknight** [comment]
LGTM. Tested again with the browser-extension and the production Chrome settings file.

File: src/sidebar/app.js (hunk 2 above)
Comment (sean-roberts) [inline src/sidebar/app.js]: yeah, this is consistent with the rest of the file as it is right now. So I am going to leave that up to eslint --fix later on

**Pull request merged**